We will do the phylogeny using a fasta output from Stacks.

Notice that when we ran populations we used an option called: "--fasta-samples". This is our dataset over a fasta file. First, we will need to split this file:

```
# This script can be found in the info folder.
perl Split_STACKS_fasta_file.pl populations.samples.fa
```

This will generate thousands of fasta files that look like this:
Locus_1.fas
Locus_1_Allele_1.fas
Locus_1_Allele_0.fas
Locus_2.fas
Locus_2_Allele_1.fas
Locus_2_Allele_0.fas

We can remove the _1.fas and the _0.fas:
```
rm *_Allele[0-1].fas
```

We then concatenate all these files together. They may need tons of memory, so be sure to feed it tons of CPU memory.

```
perl FASconCAT-G_v1.04.pl -s -p -n

# Let's move the files generated by FASconCAT above.
mv FcC* ../
```

Now we need to format a partitions dataset, and run the tree:

```
# So, we start with the FcC_info.xls file.
# I used the following command:
cat FcC_info.xls | grep "Locus" | awk '{print $1" = "$2"-"$3";"}' | sed 's/^/\tcharset /' | sed '1 i\begin sets;' | sed '1 i\#nexus' | sed '$ a\end;' > partitions.nex

# Command explained
# cat # to open the file
# grep "Locus" # to get the lines only starting with Locus
# awk '{print $1"\t"$2"-"$3";"}' # to format the file with <Locus><tab><start_pos><minus><end_pos><;>
# sed 's/^/\t/' # put a tab in the beginning
# sed '1 i\begin sets;' | 'sed '1 i\#nexus' | sed '$ a\end;' # add "begin sets", "nexus" and "end;"

# and the tree
iqtree -s FcC_supermatrix.fas -spp partitions.nex -nt 30 -safe
```


Now, we run BaCoCa:
```
# Bacoca needs differently formated partition files.
sed 's/^\tcharset //; s/;//; 2d; 1d; $d' partitions.nex > partitions.bacoca.tsv

perl BaCoCa.v1.105.pl -i FcC_supermatrix.fas -p partitions.bacoca.tsv
```


Now we plot BaCoCa's results on the tree.

1. BaCoCa provides negative and positive missingness overlaps
We have to fill the dataframe with NAs and not empty runs. You can do it by loading the file onto on Excel and:
"Control+G", "special", "select blanks", type "NA" and Control+enter

2. Then obtain a row including missingness by each specimen by summing the rows on excel from the positive overlap file.
3. We load things on R, using saga because my local version of R messes up.

```
R

library(plyr)
library(ape)
library(ggtree)
library(treeio)
library(tidyverse)
library(phytools)

## https://yulab-smu.github.io/treedata-book/chapter4.html

## My data:

tree <- read.tree("./partitions.nex.treefile")
svl <- read.table("./missingness.jc.txt", header=T,
                row.names=1)
svl <- as.matrix(svl)[,1]
fit <- phytools::fastAnc(tree,svl,vars=TRUE,CI=TRUE)

td <- data.frame(node = nodeid(tree, names(svl)),
                 trait = svl)
                 nd <- data.frame(node = names(fit$ace), trait = fit$ace)
d <- rbind(td, nd)
d$node <- as.numeric(d$node)

tree <- tidytree::full_join(tree, d, by = 'node')
pdf("stygo.pdf")
ggtree(tree, aes(color=trait), continuous = TRUE) +
    scale_color_viridis_c() + theme_minimal() + geom_tiplab()
dev.off()
```

Now, for the UPGMA tree we do the following:
```
setwd("C:/Users/Cerca/Desktop/ongoing_stuff_ambientedetrabalho/project3/10_phylogenomics/11_UPGMAtree")

library(phangorn)
library(ggtree)
library(treeio)
library(tidyverse)
library(phytools)


# results from BaCoCa
matrix<-read.table("./results_missing_data_positive_overlap_FcC_supermatrix.fas_0.txt", header=T, row.names=1, sep="\t")

tree<-upgma(matrix, method="average")
plot(tree)


windows(height=40,width=8)

ggtree(tree, colour='steelblue') +
  scale_color_viridis_c() + theme_minimal()  + geom_tiplab()

ggsave(filename = "tree.pdf" , device = "pdf", width = 15, height = 30 , units = "in" , limitsize = FALSE)

```

For the consensus tree, included as supplementary figure 7, I did:

```
# First, we get a list of specimens
grep ">" FcC_supermatrix.fas | sed "s/_All.*//" | sort > specimen_list.tsv
sed -i "s/>//" specimen_list.tsv

# And we separate these.
while read gene; do grep -A 1 --no-group-separator $gene FcC_supermatrix.fas > $gene.fas; done < specimen_list.tsv

for i in *fas; do consambig -sequence $i -outseq ${i%.fas}.consensus.fasta; sed -i "s/>.*/>${i%.fas}/" ${i%.fas}.consensus.fasta; mv ${i%.fas}.consensus.fasta ../02_ConsensusSeq/; done

# Now, we concatenate, and run the tree
perl FASconCAT-G_v1.04.pl -s -p -n

# Notice, the partitions.nex file is the same as the one generated above.

iqtree -s FcC_supermatrix.fas -spp partitions.nex -nt 30 -safe
```
